testsuite/MDAnalysisTests/analysis/test_base.py

```diff
@@ -197,6 +197,45 @@ def test_start_stop_step(u, run_kwargs, frames):
     assert_almost_equal(an.times, frames+1, decimal=4, err_msg=TIMES_ERR)
 
 
+@pytest.mark.parametrize('run_kwargs, frames', [
+    ({'frames': [4, 5, 6, 7, 8, 9]}, np.arange(4, 10)),
+    ({'frames': [0, 2, 4, 6, 8]}, np.arange(0, 10, 2)),
+    ({'frames': [4, 6, 8]}, np.arange(4, 10, 2)),
+    ({'frames': [True, True, False, True, False, True, True, False, True,
+                 False]}, (0, 1, 3, 5, 6, 8)),
+])
+def test_frame_slice(run_kwargs, frames):
+    u = mda.Universe(TPR, XTC)  # dt = 100
+    an = FrameAnalysis(u.trajectory).run(**run_kwargs)
+    assert an.n_frames == len(frames)
+    assert_equal(an.found_frames, frames)
+    assert_equal(an.frames, frames, err_msg=FRAMES_ERR)
+
+
+@pytest.mark.parametrize('run_kwargs, frames', [
+    ({'start': 4, 'frames': [4, 5, 6, 7, 8, 9]}, np.arange(4, 10)),
+    ({'stop': 6, 'frames': [0, 1, 2, 3, 4, 5]}, np.arange(0, 6)),
+    ({'step': 2, 'frames': [0, 2, 4, 6, 8]}, np.arange(0, 10, 2)),
+    ({'start': 4, 'stop': 7, 'frames': [4, 5, 6]}, np.arange(4, 8)),
+    ({'stop': 6, 'step': 2, 'frames': [0, 2, 4, 6]}, np.arange(0, 7, 2)),
+    ({'start': 4, 'step': 2, 'frames': [4, 6, 8]}, np.arange(4, 10, 2)),
+])
+def test_frame_fail(u, run_kwargs, frames):
```

Comment:
Remove the reference value that is never tested
```suggestion
def test_frame_fail(u, run_kwargs):
```

Reply:
done